### PR TITLE
Brandon/cre 211 fix agent and task config for yaml based projects

### PR DESCRIPTION
**Pull request by bhancockio**
Walk through of problem and fix:

https://www.loom.com/share/dfe23ef75a2d40268b1c26885be47e97?sid=bb6811ec-ad70-4ca6-8b77-d50cf4d15f30

### CODE REVIEW

File: src/crewai/agent.py
Comment (gvieira) [inline src/crewai/agent.py]: Just a small note: The code is pretty self-explanatory. In such cases, I try to avoid commenting. What do you think?

Comment (bhancockio) [reply src/crewai/agent.py]: Agreed! Will remove!

**Review by vinibrsl — APPROVED** [review]

File: src/crewai/agent.py
Comment (gvieira) [inline src/crewai/agent.py:120]: We do a bunch of things in `_setup_llm_callbacks` that do not require a `model_name` to be present in `model`. Should we move this check in?

File: src/crewai/agent.py
Comment (gvieira) [inline src/crewai/agent.py:131]: I wonder what's the case we are trying to prevent here. In other words, what's the scenario on which `callbacks` is not a list?

File: src/crewai/agents/agent_builder/base_agent.py
Comment (gvieira) [inline src/crewai/agents/agent_builder/base_agent.py]: How do you feel about early returns? In this case, early continue? I like it to prevent block nesting.

File: src/crewai/agents/agent_builder/base_agent.py
Comment (gvieira) [inline src/crewai/agents/agent_builder/base_agent.py]:
```suggestion
                if values.get(key) is None:
```

File: src/crewai/agents/agent_builder/base_agent.py
Comment (gvieira) [inline src/crewai/agents/agent_builder/base_agent.py]: Opposite to what I said above, I think we need a comment here. 😄 I don't know what we are doing here. lol

File: src/crewai/task.py
Comment (gvieira) [inline src/crewai/task.py]: Should these `)` be up?

File: src/crewai/task.py
Comment (gvieira) [inline src/crewai/task.py]: Hm, pretty similar to the config from above, right!? Should we centralize it somehow?

File: src/crewai/agent.py
Comment (bhancockio) [reply src/crewai/agent.py:120]: Hey! Good question.

Under the hood, we need the model name because tiktoken uses that model name to calculate tokens. 

If we don't have the model name, we can't calculate tokens.

File: src/crewai/agent.py
Comment (bhancockio) [reply src/crewai/agent.py:131]: I honestly do not know. Because the LLM class in our `BaseAgent` class is any it's hard to do any type-safety or know what's going to get passed in.

I'm happy to remove the check but it was already in place so I'm guessing someone put it there on purpose originally.

File: src/crewai/agents/agent_builder/base_agent.py
Comment (bhancockio) [reply src/crewai/agents/agent_builder/base_agent.py]: Added comment!

File: src/crewai/agents/agent_builder/base_agent.py
Comment (bhancockio) [reply src/crewai/agents/agent_builder/base_agent.py]: Refactored for early breaks with continues!

File: src/crewai/agents/agent_builder/base_agent.py
Comment (bhancockio) [reply src/crewai/agents/agent_builder/base_agent.py]: Added!

File: src/crewai/task.py
Comment (bhancockio) [reply src/crewai/task.py]: I moved up description but my formatter keeps moving the rest down.

File: src/crewai/task.py
Comment (bhancockio) [reply src/crewai/task.py]: I'll move both to a utils file.

File: src/crewai/agent.py
Comment (gvieira) [reply src/crewai/agent.py:120]: Right, but what about that thing that makes sure `callbacks` is a list, just to name one?

File: src/crewai/agent.py
Comment (gvieira) [reply src/crewai/agent.py:131]: What do you think about creating some quick-and-dirty tasks on Linear, so we can have such discussions and not block the PR? I'm sure João would have an answer for that, but let's leave it to later.

Comment (bhancockio) [reply src/crewai/agent.py:131]: Created!

https://linear.app/crewai/issue/CRE-213/llmcallbacks-list-check

**Pull request merged**
